steam_community_market/market.py

```diff
@@ -274,60 +245,117 @@ def get_volume(self, name: str, app_id: int | AppID) -> int | None:
+        :rtype: str
         """
 
         if isinstance(name, str):
             return name.replace("/", "-")
 
         raise TypeError(f'"name" must be "str", not "{type(name)}".')
 
-    def price_to_float(self, value: str) -> float | None:
+    @staticmethod
+    def _price_to_float(value: str) -> Optional[float]:
         """Converts a price from :class:`str` to :class:`float`
 
-        :param value: A price
-        :type value: :class:`str`
-        :return: :class:`float`
-        :rtype: :class:`float`, :class:`None`
+        .. versionchanged:: 1.3.0
+           * Added better parsing for strings using Regex.
+           * Added support for price values with a comma as the decimal delimiter.
 
-        .. versionchanged:: 1.2.4
+        :param value: A price in string format.
+        :type value: str
+        :return: The price in float format.
+        :rtype: float or None
         """
 
-        if not (match := re.search(r"\d{1,3}(?:[\.,]\d{1,3})*([\.,])\d{2}", value)):
+        if not (match := re.search(r"\d{1,3}(?:([\.,])\d{1,3})*(?:\1\d{2})?", value)):
             return None
 
         delimiter = match[1]
         return (
-            float(match.group())
+            float(match[0].replace(",", ""))
             if delimiter == "."
-            else float(match.group().replace(".", "").replace(",", "."))
+            else float(match[0].replace(".", "").replace(",", "."))
         )
+
+    @staticmethod
+    def _supported_currency(
+        currency: Union[int, str, SteamCurrency, SteamLegacyCurrency],
+    ) -> SteamCurrency:
+        """Returns a supported currency.
+
+        .. versionadded:: 1.3.0
+
+        :param currency: Currency used for prices.
+        :type currency: int or str or SteamCurrency or SteamLegacyCurrency
+        :raises InvalidCurrencyException: Raised when a currency is considered to be invalid by the Steam Community Market.
+        :raises LegacyCurrencyException: Raised when a currency is not supported by the Steam Community Market anymore.
+        :return: A supported currency.
+        :rtype: SteamCurrency
+        """
+
+        result: Union[SteamCurrency, None] = None
+        if isinstance(currency, int) and currency in SteamCurrency.__members__.values():
+            result = SteamCurrency(currency)
+
+        elif (
+            isinstance(currency, str)
+            and (currency := currency.upper()) in SteamCurrency.__members__.keys()
+        ):
+            result = SteamCurrency[currency]
+
+        elif isinstance(currency, SteamCurrency):
+            result = currency
+
+        if result is None:
+            raise InvalidCurrencyException(currency)
+
+        elif (
+            isinstance(currency, SteamLegacyCurrency)
+            or result.name in SteamLegacyCurrency.__members__.keys()
+            or result.value in SteamLegacyCurrency.__members__.values()
+        ):
+            raise LegacyCurrencyException(currency)
+
+        return result

```

Comment:
```suggestion
        if isinstance(currency, int) and currency in SteamCurrency.__members__.values():
            return SteamCurrency(currency)

        elif (
            isinstance(currency, str)
            and (currency := currency.upper()) in SteamCurrency.__members__.keys()
        ):
            return SteamCurrency[currency]

        elif isinstance(currency, SteamCurrency):
            return currency

        raise LegacyCurrencyException(currency)

```

Reply:
You have miss-interpreted the two exceptions and when they should get raised:
- InvalidCurrencyException gets raised when the user inputs a non-existent currency, that couldn't be supported by Steam.
- LegacyCurrencyException gets raised when the user inputs one of the legacy currencies, that are no longer supported by Steam:
  - 33 - SEK / Swedish Krona
  - 36 - BYN / Belarusian Ruble
  - 42 - BGN / Bulgarian Lev
  - 43 - HRK / Croatian Kuna
  - 44 - CZK / Czech Koruna
  - 45 - DKK / Danish Krone
  - 46 - HUF / Hungarian Forint
  - 47 - RON / Romanian Leu

Reply:
Oh sorry you're right but, what's the point even accepting legacy currencies if they just raise?

Reply:
Mainly.. Because I aspire for this library to get as competent as possible in interpreting everything related to the Steam Community Market, so it seems only fair to notify the user that the currency he's trying to use is no longer supported by the platform via an exception.